python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9059,3 +9059,43 @@ def test_dataframe_add_suffix():
     expected = pdf.add_suffix("_item")
 
     assert_eq(got, expected)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.random.normal(-100, 100, (50, 50)),
+        np.random.randint(-50, 50, (25, 30)),
+        np.random.random_sample((4, 4)),
+        np.random.uniform(10.5, 75.5, (10, 6)),
+        np.array([1.123, 2.343, 5.890, 0.0]),
+        [np.nan, None, np.nan, None],
+        {"a": [1.123, 2.343, np.nan, np.nan], "b": [None, 3, 9.08, None]},
+    ],
+)
+@pytest.mark.parametrize("periods", range(-4, 5))
+def test_diff_dataframe(data, periods):
+    gdf = cudf.DataFrame(data)
+    pdf = gdf.to_pandas()
+
+    actual = gdf.diff(periods=periods, axis=0)
+    expected = pdf.diff(periods=periods, axis=0)
+
+    assert_eq(
+        expected, actual, check_dtype=False,
+    )
+
+
+def test_diff_dataframe_invalid_axis():
+    with pytest.raises(NotImplementedError, match="Only axis=0 is supported."):
+        gdf = cudf.DataFrame(np.random.random_sample((4, 4)))
+        gdf.diff(periods=1, axis=1)
+
+
+def test_diff_dataframe_non_numeric_dtypes():
+    with pytest.raises(
+        NotImplementedError,
+        match="Diff currently only supports numeric dtypes",
+    ):
+        gdf = cudf.DataFrame(["a", "b", "c", "d", "e"])
```

Comment:
Add one more test case of mixing non-numeric and numeric types.

Reply:
```
gdf = cudf.DataFrame([1, "b", 2, "d", "e"])
gdf
```
outputs this error message: `ArrowInvalid: Could not convert 'b' with type str: tried to convert to int`

Reply:
@brandon-b-miller , is this expected behavior in cudf? i see that pandas allows df with numerical and non-numeric types..
Should i be testing for this?

Reply:
I think the problem here is just the call to the dataframe constructor. Right now it's just passing a heterogeneous list. I think maybe what you want here is a dataframe with columns of different dtypes. You could do something like:

```python
gdf = cudf.DataFrame({
    'int_col': [1,2,3,4,5],
    'float_col': [1.0,2.0,3.0,4.0,5.0],
    'string_col':['a','b','c','d','e']
})
```

Reply:
gotcha, thanks for the explanation 👍🏾